lua/session_manager/init.lua

```diff
@@ -68,13 +74,24 @@ function session_manager.save_current_session()
   end
 end
 
+local autoloaders = {
+  [AutoloadMode.Disabled] = function() return true end,
+  [AutoloadMode.CurrentDir] = session_manager.load_current_dir_session,
+  [AutoloadMode.LastSession] = session_manager.load_last_session,
+}
+
 --- Loads a session based on settings. Executed after starting the editor.
 function session_manager.autoload_session()
-  if config.autoload_mode ~= AutoloadMode.Disabled and vim.fn.argc() == 0 and not vim.g.started_with_stdin then
-    if config.autoload_mode == AutoloadMode.CurrentDir then
-      session_manager.load_current_dir_session()
-    elseif config.autoload_mode == AutoloadMode.LastSession then
-      session_manager.load_last_session()
+  if vim.fn.argc() > 0 or vim.g.started_with_stdin then
+    return
+  end
+  local modes = config.autoload_mode
+  if not vim.isarray(config.autoload_mode) then
+    modes = { config.autoload_mode }
+  end
```

Comment:
Let's put newlines around it. Just a personal preference :)
```suggestion

  local modes = config.autoload_mode
  if not vim.isarray(config.autoload_mode) then
    modes = { config.autoload_mode }
  end

```